D8AGE-871: Fix the callback URLs in the translation request.

intelektron opened the pull request:
## D8AGE-871

### Description

Right now, the callbacks in the `modules/oe_translation_cdt/src/Mapper/TranslationRequestMapper.php:173` file are assigned to wrong routes. The `oe_translation_cdt.request_status_callback` should be assigned to `$request_status_url`

intelektron commented on modules/oe_translation_cdt/modules/oe_translation_cdt_mock/oe_translation_cdt_mock.install:
```diff
@@ -0,0 +1,22 @@
+<?php
```
I also added this change to make testing a bit more reliable, previously the mock module would throw an error if the old token was still valid.

drishu commented on modules/oe_translation_cdt/src/Mapper/TranslationRequestMapper.php:
```diff
@@ -175,14 +175,14 @@ protected function countVolume(int $character_count): float {
     // Set the LANGUAGE_NOT_SPECIFIED to avoid the language suffix in the URL.
     // @see \Drupal\language\Plugin\LanguageNegotiation\LanguageNegotiationUrl::processOutbound()
-    $job_status_url = Url::fromRoute(
+    $request_status_url = Url::fromRoute(
```
Cant belive we missed it, lets improve also in TranslationRequestMapperTest the testing of getCallbackBaseUrl

intelektron replied:
True! I've just added some additional asserts.

drishu reviewed (changes requested):
Not sure since when, but drone seems to fail in cloning the cdt client

intelektron commented:
I rebased the epic onto latest 2.x, and fixed the remaining GrumPHP issues. We will be deploying the module with R29 anyway, and by doing this we get rid of Drone problems.
